src/libostree/ostree-repo-commit.c

```diff
@@ -598,27 +598,30 @@ create_regular_tmpfile_linkable_with_content (OstreeRepo *self,
 }
 
 static gboolean
-_check_support_reflink (OstreeRepo *self, gboolean *supported, GError **error)
+_check_support_reflink (OstreeRepo *src, OstreeRepo *dest, gboolean *supported, GError **error)
 {
-  /* We have not checked yet if the file system supports reflinks, do it here */
-  if (g_atomic_int_get (&self->fs_support_reflink) == 0)
+  /* The two repositories are on different devices */
+  if (src->device != dest->device)
```

Comment:
Since this is the only place where we compare them, maybe it's simplest to move this condition down to the two places that call this with different src,dest?